python/cudf/cudf/core/dataframe.py

```diff
@@ -3785,6 +3785,60 @@ def to_csv(
             chunksize,
         )
 
+    def scatter_by_map(self, map_index, map_size=None):
+        """Scatter to a list of dataframes.
+
+        Uses map_index to determine the destination
+        of each row of the original DataFrame.
+
+        Parameters
+        ----------
+        map_index : Series, str or list-like
+            Scatter assignment for each row
+        map_size : int
+            Length of output list. Must be >= uniques in map_index
+
+        Returns
+        -------
+        A list of cudf.DataFrame objects.
+        """
+
+        # map_index might be a column name or array,
+        # make it a Series
+        if isinstance(map_index, str):
+            map_index = self[map_index]
+        else:
+            map_index = Series(map_index)
+
+        # Convert float to integer
+        if map_index.dtype == np.float:
+            map_index = map_index.astype(np.int64)
```

Comment:
I believe you'll want an int32 here instead of an int64.

@shwina did we ever build the utility for building a standardized map?

Reply:
libcudf's `scatter` and `gather` are being updated to accept a column (of any integer dtype and including negative values), so there won't be need for one...